src/Profiles.ts

```diff
@@ -336,6 +338,178 @@ export class Profiles {
+
+            let deleteLabel: string;
+            let deletedProfile: IProfileLoaded;
+            if (!node){
+                deletedProfile = await this.getDeleteProfile();
+                if (!deletedProfile) {
+                    return;
+                } else {
+                    deleteLabel = deletedProfile.name;
+                }
+            } else {
+                deletedProfile = node.getProfile();
+                deleteLabel = node.getProfileName();
+            }
+            const deleteSuccess = await this.deletePrompt(deletedProfile);
+            if (!deleteSuccess){
+                vscode.window.showInformationMessage(localize("deleteProfile.noSelected",
+                    "Operation Cancelled"));
+                return;
+            }
+
+            // Delete from Data Set Tree
+            datasetTree.mSessionNodes.forEach((sessNode) => {
+                if (sessNode.getProfileName() === deleteLabel) {
+                    datasetTree.deleteSession(sessNode);
+                }
+            });
+            await datasetTree.refresh();
+
+            // Delete from Data Set Favorites
+            datasetTree.mFavorites.forEach((ses) => {
+                const findNode = ses.label.substring(1, ses.label.indexOf("]")).trim();
+                if (findNode === deleteLabel) {
+                    datasetTree.removeFavorite(ses);
+                }
+            });
+            await datasetTree.refresh();
+
+            // Delete from Data Set Recall
+            const recallDs: string[] = datasetTree.getRecall();
+            for (let i = recallDs.length - 1; i >= 0; i--) {
+                const findNode = recallDs[i].substring(1, recallDs[i].indexOf("]")).trim();
+                if (findNode === deleteLabel) {
+                    datasetTree.removeRecall(recallDs[i]);
+                }
+            }
+
+            await datasetTree.refresh();
+
+            // Delete from USS Tree
+            ussTree.mSessionNodes.forEach((sessNode) => {
+                if (sessNode.getProfileName() === deleteLabel) {
+                    ussTree.deleteSession(sessNode);
+                }
+            });
+
+            // Delete from USS Favorites
+            ussTree.mFavorites.forEach((ses) => {
+                const findNode = ses.label.substring(1, ses.label.indexOf("]")).trim();
+                if (findNode === deleteLabel) {
+                    ussTree.removeFavorite(ses);
+                }
+            });
+            await ussTree.refresh();
+
+            // Delete from USS Recall
+            const recallUSS: string[] = ussTree.getRecall();
+            for (let i = recallUSS.length - 1; i >= 0; i--) {
+                const findNode = recallUSS[i].substring(1, recallUSS[i].indexOf("]")).trim();
+                if (findNode === deleteLabel) {
+                    ussTree.removeRecall(recallUSS[i]);
+                }
+            }
+
+            await ussTree.refresh();
```

Comment:
Similarly to the datset refreshes - would one ussTree refresh ad the end suffice?